distill/process/log.py

```diff
@@ -0,0 +1,83 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from pydantic import BaseModel, Field, ValidationError
+from pydantic.config import ConfigDict
+
+from decimal import Decimal
+from typing import List, Union, Any
+from enum import Enum
+
+class Browser(BaseModel):
+    browser: str
+    version: str
+
+class Location(BaseModel):
+    x: Union[int, None]
+    y: Union[int, None]
+
+class ScrnRes(BaseModel):
+    width: int
+    height: int 
+
+class LogType(str, Enum):
+    raw = "raw"
+    custom = "custom"
+
+class Details(BaseModel):
+    window: bool
+
+class LogSchema(BaseModel):
+    """
+    A raw or custom log produced by userale
+    """
+    model_config = ConfigDict(title="Log")
```

Comment:
We should also use an [alias generator](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.alias_generator) to handle camel case <---> snake case conversions.

All our field names in the python class should be snake case. We should not mix cases. However, the incoming fields will be camel case since they're coming from TS/JS/JSON.

When we dump the data back out, we want it to auto-convert from snake case to camel case.